lib/archive/producer.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"archive/tar"
+	"io"
+
+	"github.com/go-openapi/runtime"
+)
+
+func TarProducer() runtime.Producer {
+	return runtime.ProducerFunc(func(writer io.Writer, data interface{}) error {
+		return runtime.ByteStreamProducer().Produce(tar.NewWriter(writer), data)
```

Comment:
this is not how we are processing tars at the moment. Just return a bytestream producer.